contracts/src/utils/pausable.rs

```diff
@@ -68,73 +72,75 @@ impl Pausable {
         self._paused.get()
     }
 
-    /// Triggers `Paused` state.
+    /// Modifier to make a function callable only when the contract is NOT
+    /// paused.
     ///
     /// # Arguments
     ///
-    /// * `&mut self` - Write access to the contract's state.
+    /// * `&self` - Read access to the contract's state.
     ///
     /// # Errors
     ///
-    /// If the contract is in `Paused` state, then the error
+    /// If the contract is in the `Paused` state, then the error
     /// [`Error::EnforcedPause`] is returned.
-    pub fn pause(&mut self) -> Result<(), Error> {
-        self.when_not_paused()?;
-        self._paused.set(true);
-        evm::log(Paused { account: msg::sender() });
+    pub fn when_not_paused(&self) -> Result<(), Error> {
+        if self._paused.get() {
+            return Err(Error::EnforcedPause(EnforcedPause {}));
+        }
         Ok(())
     }
 
-    /// Triggers `Unpaused` state.
+    /// Modifier to make a function callable
+    /// only when the contract is paused.
     ///
     /// # Arguments
     ///
-    /// * `&mut self` - Write access to the contract's state.
+    /// * `&self` - Read access to the contract's state.
     ///
     /// # Errors
     ///
     /// If the contract is in `Unpaused` state, then the error
     /// [`Error::ExpectedPause`] is returned.
-    pub fn unpause(&mut self) -> Result<(), Error> {
-        self.when_paused()?;
-        self._paused.set(false);
-        evm::log(Unpaused { account: msg::sender() });
+    pub fn when_paused(&self) -> Result<(), Error> {
```

Comment:
Same question as above.
I know we already had this as part of the contract public abi, but do we want this? It seems like it would be better to move it and keep it internal to the contract.

Reply:
It's a good question. IMHO modifiers do not need to be callable so I would move them from `#[public]` impl block to the second one.

Reply:
Made them private and updated tests